modules/base/rendering/screenspacerenderablerenderable.cpp

```diff
@@ -0,0 +1,294 @@
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/screenspacerenderablerenderable.h>
+
+#include <openspace/camera/camera.h>
+#include <openspace/rendering/renderable.h>
+#include <openspace/scene/rotation.h>
+#include <openspace/scene/scale.h>
+#include <openspace/scene/translation.h>
+#include <openspace/util/updatestructures.h>
+
+namespace {
+    constexpr openspace::properties::Property::PropertyInfo TimeInfo = {
+        "Time",
+        "Time",
+        "The time (in J2000 seconds) that is used to calculate transformations and the "
+        "renderable's data."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CameraPositionInfo = {
+        "CameraPosition",
+        "Camera Position",
+        "Specifies the location of the virtual camera that is showing the renderable "
+        "class. This position is provided in meters."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CameraCenterInfo = {
+        "CameraCenter",
+        "Camera Center",
+        "The location of the camera's focal point. The camera's view direction will "
+        "always be pointing at the provided center location. This position is provided "
+        "in meters."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CameraUpInfo = {
+        "CameraUp",
+        "Camera Up",
+        "The direction that is 'up' for the provided camera. This value does not have "
+        "any units."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CameraFovInfo = {
+        "CameraFov",
+        "Camera Field-of-view",
+        "The camera's field-of-view in degrees."
+    };
+
+    const openspace::properties::PropertyOwner::PropertyOwnerInfo TransformInfo = {
+        "Transform",
+        "Transform",
+        "The Translation, Rotation, and Scale that are applied to the rendered "
+        "Renderable."
+    };
+
+    struct [[codegen::Dictionary(ScreenSpaceRenderableRenderable)]] Parameters {
```

Comment:
Please add a description above the struct as well that explains the basics of this class :)  

Maybe it's worth mentioning that it creates a framebuffer "window" to draw the object in?  To me, that's the main weirdness about this screenspace renderable type. It could also be used to recommend users to set the scale to something larger than default. 

Maybe we should have an example that just sets the size of the rendered output, to make this extra clear?

Reply:
I turned the `size` property of the ScreenSpaceFramebuffer into a Vec2 since we didn't really need the offset in x and y anyway and it should be a hopefully bit clearer now.

Basically for all other ScreenSpace objects we get a native resolution prescribed (image resolution or video resolution), but for these we need to manually specify the size of the object.
The confusion is, I guess, that we have both the `size` + `scale` wheresa the other have the `size` portion hidden since it is taken from the source material

Reply:
Ahh, I see. For me, the confusing part is that the object will "clip" if it's rendered outside the frame, but I guess that can be fixed either by setting the size or the scale. 

But this is probably clear enough :)